delphi-frontend/src/main/java/org/sonar/plugins/communitydelphi/api/reporting/FixEdit.java

```diff
@@ -0,0 +1,206 @@
+  }
+
+  public static FixEdit insertBefore(String text, DelphiNode node) {
+    return insert(text, node.getBeginLine(), node.getBeginColumn());
+  }
+
+  public static FixEdit insertAfter(String text, DelphiNode node) {
+    return insert(text, node.getEndLine(), node.getEndColumn());
+  }
+
+  public static FixEdit copy(DelphiNode node, FilePosition location) {
+    return new NodeCopyFixEdit(node, location);
+  }
+
+  public static FixEdit copyBefore(DelphiNode node, DelphiNode referenceNode) {
+    return copy(
+        node,
+        FilePosition.from(
+            referenceNode.getBeginLine(),
+            referenceNode.getBeginColumn(),
+            referenceNode.getBeginLine(),
+            referenceNode.getBeginColumn()));
+  }
+
+  public static FixEdit copyAfter(DelphiNode node, DelphiNode referenceNode) {
+    return copy(
+        node,
+        FilePosition.from(
+            referenceNode.getEndLine(),
+            referenceNode.getEndColumn(),
+            referenceNode.getEndLine(),
+            referenceNode.getEndColumn()));
+  }
+
+  public static FixEdit copyReplacing(DelphiNode node, DelphiNode referenceNode) {
+    return copy(node, FilePosition.from(referenceNode));
+  }
+
+  public static FixEdit move(DelphiNode node, FilePosition location) {
+    return new NodeMoveFixEdit(node, location);
+  }
+
+  public static FixEdit moveBefore(DelphiNode node, DelphiNode referenceNode) {
+    return move(
+        node,
+        FilePosition.from(
+            referenceNode.getBeginLine(),
+            referenceNode.getBeginColumn(),
+            referenceNode.getBeginLine(),
+            referenceNode.getBeginColumn()));
+  }
+
+  public static FixEdit moveAfter(DelphiNode node, DelphiNode referenceNode) {
+    return move(
+        node,
+        FilePosition.from(
+            referenceNode.getEndLine(),
+            referenceNode.getEndColumn(),
+            referenceNode.getEndLine(),
+            referenceNode.getEndColumn()));
+  }
+
+  public static FixEdit moveReplacing(DelphiNode node, DelphiNode referenceNode) {
+    return new NodeMoveFixEdit(node, FilePosition.from(referenceNode));
+  }
+
+  public static FixEdit replace(DelphiNode node, String text) {
+    return replace(FilePosition.from(node), text);
+  }
+
+  public static FixEdit replace(FilePosition range, String text) {
+    return new ReplaceFixEdit(range, text);
+  }
+
+  public static FixEdit delete(DelphiNode node) {
+    return delete(FilePosition.from(node));
+  }
+
+  public static FixEdit delete(FilePosition range) {
+    return new DeleteFixEdit(range);
+  }
+
+  private static CommonToken getAntlrToken(DelphiToken token) {
+    return (CommonToken) ((DelphiTokenImpl) token).getAntlrToken();
+  }
+
+  private static String getNodeText(DelphiNode node) {
+    int startIndex = getAntlrToken(node.getFirstToken()).getStartIndex();
+    int endIndex = getAntlrToken(node.getLastToken()).getStopIndex();
+
+    try {
```

Comment:
I think we want a try-with-resources here for the stream.

Reply:
`DelphiFileStream` does not actually hold open a file stream and so doesn't have a `close()`. That being said, it's pretty suboptimal to open and read an entire file over and over again for every quick fix, so I've rewritten this code to share a memoized stream across all edits in a file.